telemetry/definitions/commonDefinitions.json

```diff
@@ -6860,6 +6862,44 @@
                 }
             ]
         },
+        {
+            "name": "toolkit_closeModule",
+            "description": "The user closed 'something' (specified by 'module'). Examples: a view, feature, resource, ...",
+            "metadata": [
+                {
+                    "type": "module",
+                    "required": true
+                }
+            ],
+            "passive": true
+        },
+        {
+            "name": "toolkit_featureState",
+            "description": "Represents the current enabled state of a feature. Used to track user journey through a feature. Emitted after feature-specific operations of interest in the Toolkit.",
```

Comment:
how does this differ from `aws_featureConfig` which was recently added? https://github.com/aws/aws-toolkit-common/blob/940ac45eeb76a5a6bc6b92c8a66306a0d1f363b7/telemetry/definitions/commonDefinitions.json#L2701C22-L2703

`aws_featureConfig` mentions AB testing but i don't see why it needs to be restricted to that. @ivikash

Reply:
if this is truly different, would be helpful for the description to mention the differences

Reply:
I did consider using `aws_featureConfig`. My understanding of this metric is:
- it is a statement of immutable fact around a feature ("Toolkit has been told that for A/B purposes, feature X is A or B")
- it is likely not frequently emitted

The intent behind `toolkit_featureState` is:
- emitted often, representing the user's current state as they click or interact their way through a workflow
- the state is mutable (it is possible for User and/or Toolkit to affect the enabled state)
- indicates why state is being emitted ("User has specified that feature is enabled" or "User is entering this workflow, here's the starting value")

Based on the requirements I discussed with @dhruvigajjar , `aws_featureConfig` did not match the needs.

Reply:
> representing the user's current state as they click or interact their way through a workflow

This "journey through a feature" sounds like it could live on the existing feature-related metrics. Seems like we're inverting things with `toolkit_featureState`: now the "feature" will live as a field on `toolkit_featureState`, even though each "feature" already has dedicated metrics (which could themselves have a `featureId` and related fields).

Feels like a denormalization

Reply:
I think that's okay. If we had done this in the past, we may have used it more. I'm not inclined to make metrics with "gettingStarted" in the name, that is how we continue to proliferate metrics definitions. Not that that is a bad thing, but fewer is easier to manage.

Reply:
`featureConfig` has lot of fields which are only specific to AB testing which might be confusing during interpretation.
Since VS's getting started page has purpose of giving user feature specific (AWS Toolkit or AmazonQ)experience I thought dedicated metric for user's experience is `enabled` or not would be helpful, thus `toolkit_featureState`. 

I see this as equivalent of VSC/JB's `auth_addConnection` (as part of getting started page/auth page) that gives information about user's end-to-end experience through auth flow,`toolkit_featureState` will give information about user's journey through enabling a feature.